contrib/python/src/python/pants/contrib/python/checks/tasks/checkstyle/checkstyle.py

```diff
@@ -164,22 +164,15 @@ def checkstyle(self, interpreter, sources):
   def execute(self):
     """"Run Checkstyle on all found non-synthetic source files."""
 
-    # If we are linting for Python 3, skip lint altogether.
-    # Long-term Python 3 linting solution tracked by:
-    # https://github.com/pantsbuild/pants/issues/5764
-    interpreter = self.context.products.get_data(PythonInterpreter)
-    # Check interpreter is not 'None' for test cases that do not
-    # run the python interpreter selection task.
-    if interpreter and interpreter.version >= (3, 0, 0):
-      self.context.log.info('Linting is currently disabled for Python 3 targets.\n '
-                            'See https://github.com/pantsbuild/pants/issues/5764 for '
-                            'long-term solution tracking.')
-      return
+    tgts_by_compatibility, _ = interpreter_cache.partition_targets_by_compatibility(python_tgts)
```

Comment:
You should kill the `prepare` method override now - you get the appropriate interpreters directly from the interpreter_cache now. And that interpreter_cache will need to be build here somewhere using the PythonSetup and PythonRepos subsystem dependencies declared above.